Automattic/simplenote-ios
# Fix issue where unit test log out simulator

### Fix
This PR fixes #1025 
Currently when you run the unit tests in Simplenote iOS, the simulator get's logged out of the app.  This is happening because the KeychainMigratorTests are making calls to the username field, which has a setter that updates the Userdefaults username value.  In the tests they get set to nil, effectively logging the sim out.

<img src="https://cdn-std.droplr.net/files/acc_593859/wzBl2E" />

This PR changes KeychainMigrator to hold onto a local reference of UserDefaults instead, so at testing we can create a KeychainMigrator and a test UserDefaults, leaving the simulator unaffected

### Test
1. Log into a simplenote account with any simulator
2. run the full unit test suite for simplenote ios
3. check and make sure that all tests pass
4. check and make sure that the simulator is still logged in

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
> These changes do not require release notes.
